### PR TITLE
Improve Deletion of Deploymentconfig

### PR DESCRIPTION
**Pull request by prsurve**
ocs-ci fails if called deletion of dc for singe dc twice

Signed-off-by: prsurve <prsurve@redhat.com>

### CODE REVIEW

**Comment by prsurve** [comment]
### Before
`18:24:38 - MainThread - tests.manage.test - INFO - Deleting resource
18:24:38 - MainThread - tests.sanity_helpers - INFO - Deleting resources as a sanity functional validation
18:24:38 - MainThread - tests.sanity_helpers - INFO - First deletion
18:24:38 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig delete deploymentconfig pod-test-rbd-e2b401b73c584d41858eb28473a9bbcc
18:24:40 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig get deploymentconfig pod-test-rbd-e2b401b73c584d41858eb28473a9bbcc -o yaml
18:24:42 - MainThread - ocs_ci.ocs.ocp - INFO - deploymentconfig pod-test-rbd-e2b401b73c584d41858eb28473a9bbcc got deleted successfully
18:24:42 - MainThread - tests.sanity_helpers - INFO - Second deletion
18:24:42 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig delete deploymentconfig pod-test-rbd-e2b401b73c584d41858eb28473a9bbcc
FAILED                                                                                                                                                                                                                                 [100%]
`
### After
`18:33:20 - MainThread - tests.manage.test - INFO - Deleting resource
18:33:20 - MainThread - tests.sanity_helpers - INFO - Deleting resources as a sanity functional validation
18:33:20 - MainThread - tests.sanity_helpers - INFO - First deletion
18:33:20 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig get deploymentconfig  -n namespace-test-cd6f27354acc4a608bafb24a427a1195 -o yaml
18:33:22 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig delete deploymentconfig pod-test-rbd-19810d350c644093b8425a9f2324cacf
18:33:24 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig get deploymentconfig pod-test-rbd-19810d350c644093b8425a9f2324cacf -n namespace-test-cd6f27354acc4a608bafb24a427a1195 -o yaml
18:33:27 - MainThread - ocs_ci.ocs.ocp - INFO - deploymentconfig pod-test-rbd-19810d350c644093b8425a9f2324cacf got deleted successfully
18:33:27 - MainThread - tests.sanity_helpers - INFO - Second deletion
18:33:27 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig get deploymentconfig  -n namespace-test-cd6f27354acc4a608bafb24a427a1195 -o yaml
18:33:28 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig delete PersistentVolumeClaim pvc-test-6112f57e0baf4724bb9755e49b9c5852
18:33:59 - MainThread - ocs_ci.utility.utils - INFO - Executing command: oc -n namespace-test-cd6f27354acc4a608bafb24a427a1195 --kubeconfig /home/prsurve/OCS4/tmp/auth/kubeconfig get PersistentVolumeClaim pvc-test-6112f57e0baf4724bb9755e49b9c5852 -n namespace-test-cd6f27354acc4a608bafb24a427a1195 -o yaml
18:34:02 - MainThread - ocs_ci.ocs.ocp - INFO - PersistentVolumeClaim pvc-test-6112f57e0baf4724bb9755e49b9c5852 got deleted successfully
PASSED                                                                                                                                                                                                                                 [100%]
------------------------------------------------------------------------------------------------------------- live log teardown -------------------------------------------------------------------------------------------------------------
`

**Comment by prsurve** [comment]
Verified Locally

File: tests/helpers.py [hunk 1]
```diff
@@ -1165,9 +1165,13 @@ def delete_deploymentconfig(pod_obj):
...
-    dc_ocp_obj = ocp.OCP(kind=constants.DEPLOYMENTCONFIG)
-    dc_ocp_obj.delete(resource_name=pod_obj.get_labels().get('name'))
-    dc_ocp_obj.wait_for_delete(resource_name=pod_obj.get_labels().get('name'))
+    dc_ocp_obj = ocp.OCP(kind=constants.DEPLOYMENTCONFIG, namespace=pod_obj.namespace)
+    pod_data_list = dc_ocp_obj.get()['items']
+    if pod_data_list:
+        for pod_data in pod_data_list:
+            if pod_obj.get_labels().get('name') == pod_data.get('metadata').get('name'):
+                dc_ocp_obj.delete(resource_name=pod_obj.get_labels().get('name'))
+                dc_ocp_obj.wait_for_delete(resource_name=pod_obj.get_labels().get('name'))
```
Comment (vasukulkarni) [inline tests/helpers.py]: There seems to be indentation issue?

File: tests/helpers.py [hunk 2]
```diff
@@ -1165,9 +1165,13 @@ def delete_deploymentconfig(pod_obj):
     Args:
          pod_obj (object): Pod object
```
Comment (ebenahar) [inline tests/helpers.py]: I think the function name should be changed to delete_deploymentconfig_pods()

Comment (prsurve) [reply tests/helpers.py]: yes make sense will update in next commit

File: tests/helpers.py (hunk 1 above)
Comment (prsurve) [reply tests/helpers.py]: will check and update if required in next commit

Comment (prsurve) [reply tests/helpers.py]: will check and update if required in next commit

File: tests/helpers.py (hunk 2 above)
Comment (prsurve) [reply tests/helpers.py]: done

File: tests/helpers.py (hunk 1 above)
Comment (prsurve) [reply tests/helpers.py]: i checked i don't see any indentation issue here

**Pull request merged**